pyctuator/impl/pyctuator_impl.py

```diff
@@ -102,8 +102,23 @@ def get_health(self) -> HealthSummary:
             for provider in self.health_providers
             if provider.is_supported()
         }
-        service_is_up = all(health_status.status == Status.UP for health_status in health_statuses.values())
-        return HealthSummary(Status.UP if service_is_up else Status.DOWN, health_statuses)
+
+        # Health is UP if no provider is registered
+        if not health_statuses:
+            return HealthSummary(Status.UP, health_statuses)
+
+        # If there's at least one provider and any of the providers is DOWN, the service is DOWN
+        service_is_down = any(health_status.status == Status.DOWN for health_status in health_statuses.values())
```

Comment:
Nice!
🤓

Reply:
Agree, the `any` and `all` functions are really awesome.